src/de/deserializer.zig

```diff
@@ -254,8 +254,8 @@ pub fn Deserializer(comptime dbt: anytype, comptime Reader: type) type {
             const token = try self.parser.nextAlloc(ally, .alloc_if_needed);
             defer if (token == .allocated_number) ally.free(token.allocated_number);
 
-            const value = switch (token) {
-                inline .number, .allocated_number => |slice| blk: {
+            switch (token) {
+                .number, .allocated_number => |slice| {
```

Comment:
Let's keep the `inline` here. That's how `std.json` does it and I'm also not quite sure why it works without the `inline` there. `.number` and `.allocated_number` have different payload types so I thought an `inline` was required...

```suggestion
                inline .number, .allocated_number => |slice| {
```

Reply:
It works without inline because the types are `[]const u8` and `[]u8`, which both coerce to `[]const u8`. Personally, I perfer not to use inline if its not semantically necessary and rather let the compiler decide what to do. But if you'd still like to keep it just let me know and I'll change it back.